packages/Search/src/DTK_FineSearchFunctor.hpp

```diff
@@ -0,0 +1,361 @@
+    }
+};
+
+class Wedge_6
+{
+  public:
+    template <typename DeviceType>
+    KOKKOS_INLINE_FUNCTION void
+    mapToReferenceFrame( Kokkos::View<Coordinate *, DeviceType> reference_point,
+                         Kokkos::View<Coordinate *, DeviceType> physical_point,
+                         Kokkos::View<Coordinate **, DeviceType> nodes ) const
+    {
+        typedef Intrepid2::Impl::Basis_HGRAD_WEDGE_C1_FEM basis_type;
+        Intrepid2::Impl::CellTools::Serial::mapToReferenceFrame<basis_type>(
+            reference_point, physical_point, nodes );
+    }
+
+    template <typename DeviceType>
+    KOKKOS_INLINE_FUNCTION bool
+    checkPointInclusion( Kokkos::View<Coordinate *, DeviceType> reference_point,
+                         double threshold ) const
+    {
+        return Intrepid2::Impl::Wedge<6>::checkPointInclusion( reference_point,
+                                                               threshold );
+    }
+};
+
+class Wedge_18
+{
+  public:
+    template <typename DeviceType>
+    KOKKOS_INLINE_FUNCTION void
+    mapToReferenceFrame( Kokkos::View<Coordinate *, DeviceType> reference_point,
+                         Kokkos::View<Coordinate *, DeviceType> physical_point,
+                         Kokkos::View<Coordinate **, DeviceType> nodes ) const
+    {
+        typedef Intrepid2::Impl::Basis_HGRAD_WEDGE_C2_FEM basis_type;
+        Intrepid2::Impl::CellTools::Serial::mapToReferenceFrame<basis_type>(
+            reference_point, physical_point, nodes );
+    }
+
+    template <typename DeviceType>
+    KOKKOS_INLINE_FUNCTION bool
+    checkPointInclusion( Kokkos::View<Coordinate *, DeviceType> reference_point,
+                         double threshold ) const
+    {
+        return Intrepid2::Impl::Wedge<18>::checkPointInclusion( reference_point,
+                                                                threshold );
+    }
+};
+}
+
+namespace Functor
+{
+template <typename CellType, typename DeviceType>
+class FineSearch
+{
+  public:
+    FineSearch(
+        Kokkos::View<Coordinate **, DeviceType> reference_points,
+        Kokkos::View<bool *, DeviceType> point_in_cell,
+        Kokkos::View<Coordinate **, DeviceType> physical_points,
+        Kokkos::View<Coordinate ***, DeviceType> cells,
+        Kokkos::View<unsigned int *, DeviceType> coarse_search_output_points,
+        Kokkos::View<unsigned int *, DeviceType> coarse_search_output_cells );
+
+    KOKKOS_INLINE_FUNCTION
+    void operator()( unsigned int const i ) const
+    {
+        unsigned int const point_index = _coarse_search_output_points( i );
+        unsigned int const cell_index = _coarse_search_output_cells( i );
+        // Get subviews
+        Kokkos::View<Coordinate *, DeviceType> ref_point( _reference_points, i,
+                                                          Kokkos::ALL() );
+        Kokkos::View<Coordinate *, DeviceType> phys_point(
+            _physical_points, point_index, Kokkos::ALL() );
```

Comment:
But it's also a lot harder to make it compile because the type of `auto phys_point` is pretty ugly. It also make it harder to read the signature of `mapToReferenceFrame` and `checkPointInclusion`.